test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -80,6 +87,53 @@ static public void setUp() {
 			new ValueTypePoint2D(new ValueTypeInt(10), new ValueTypeInt(20))
 		};
 		vtPointAryOffset1 = vtPointAryOffset0 + arrayElementSize(vtPointAry);
+		vtIntAry = new ValueTypeInt[] { new ValueTypeInt(1), new ValueTypeInt(2) };
+		vtIntAryOffset1 = vtIntAryOffset0 + arrayElementSize(vtIntAry);
+	}
+
+	public static class CompareAndDoSomethingFunction {
+		private Method method;
+
+		public CompareAndDoSomethingFunction(String methodName) throws NoSuchMethodException {
+			this.method = myUnsafe.getClass().getMethod(
+				methodName,
+				new Class[]{Object.class, long.class, Class.class, Object.class, Object.class}
+			);
+		}
+
+		public boolean execute(Object obj, long offset, Class<?> clz, Object v1, Object v2) throws Throwable {
+			Object returned = null;
+			try {
+				returned = method.invoke(myUnsafe, obj, offset, clz, v1, v2);
+			} catch (InvocationTargetException exception) {
+				throw exception.getCause();
+			}
+			boolean result = false;
+			if (returned instanceof Boolean) {
+				result = (Boolean)returned;
+			} else {
+				result = returned == v1;
+			}
+			return result;
+		}
+
+		public String toString() {
+			return "CompareAndDoSomethingFunction(" + method.getName() + ")";
+		}
+	}
+
+	@DataProvider(name = "compareAndDoSomethingFuncs")
+	static public Object[][] compareAndDoSomethingFuncs() throws NoSuchMethodException {
+		return new Object[][] {
```

Comment:
You don't need all the wrappers here. I think `compareAndSetValue()` and `compareAndExchangeValue()` should be sufficient.

Reply:
Okay, what about when I add tests for `getAndSetValue`, `getAndSetValueRelease`, and `getAndSetValueAcquire`? All 3 functions are almost identical so I could only write tests for `getAndSetValue` and ignore the others, or I could make another data provider for these 3 functions

Reply:
Test `getAndSetValue()` should be sufficient.

Reply:
changed in 8add8f2